### ISSUE TITLE
Your example code appears to be wrong.

### ISSUE THREAD

**Issue by MarqueIV** [dotnet/docs#7038]
Reading your example code for the void return type, I'm not sure it does what you think.  If you add that to a console app and run it, you get wildly inaccurate results.  Resetting the counter in the 'event handler' (while declaring an actual event handler that has nothing to do with the code) is just confusing.  I would revisit it with a better example. It doesn't clearly show what you think it does.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d055727b-c97c-4d25-39b8-a4d3f21d0444
* Version Independent ID: c8a7c6f1-8ee1-5030-db12-3db65a968501
* Content: [Async Return Types (C#)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/async-return-types)
* Content Source: [docs/csharp/programming-guide/concepts/async/async-return-types.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/async/async-return-types.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
Thanks for pointing this out @MarqueIV 

I've added it to our backlog, and our project to improve the async docs in general.

**Comment by MarqueIV** [comment]
No worries, and thanks. I had decided to dive into the Async features and was relying on this documentation and the more I read it and tried to apply it to my code, the more confusing it got. It wasn't until I created a new project myself with that sample code that the issues with the code started to reveal itself.  In a way the time I took to unwind that code, then play around with it until it actually worked *did* help me learn it better, but unfortunately it also showed me just how far off that documentation was.  While I'm all for experimenting to learn, I can't say that should apply to the documentation too! :)
